### PR TITLE
.github/build-tools: build again verbose and single threaded after a failure

### PR DESCRIPTION
**Pull request by marc-hb**
2 commits, see commit messages.

Building in parallel is much faster but it makes logs unreadable and
build failures impossible to understand.

To show errors, try to build again with a single thread on failure.

### CODE REVIEW

**Comment by marc-hb** [comment]
Test results are in throw-way #5639 where this was combined with a test failure for demonstration purposes.

**Comment by marc-hb** [comment]
ICL is broken as often @kkarask in https://sof-ci.01.org/sof-pr-viewer/#/build/PR5640/build8993943. Maybe we should stop testing ICL?

The two failures in https://sof-ci.01.org/sofpr/PR5640/build12608/devicetest/ are unrelated #5352

**Comment by lgirdwood** [comment]
> ICL is broken as often @kkarask in https://sof-ci.01.org/sof-pr-viewer/#/build/PR5640/build8993943. Maybe we should stop testing ICL?

This one is just is usually the DUT needing a reboot.

**Review by lgirdwood — APPROVED** [review]

**Comment by kkarask** [comment]
We struggle with several issues on CI platforms now (ICL, TGL, JSL). It could take a while to solve all problems.

**Comment by wszypelt** [comment]
@marc-hb @lgirdwood 
The problem is on internal service, the issue has been reported, we are waiting for the task to be completed. Unfortunately, disconnecting the machine with the error will not solve the problem, because at this point it will appear on the next machine.

File: .github/workflows/installer.yml [hunk 1]
```diff
@@ -27,10 +27,12 @@ jobs:
...
 
       # -j3 tests the ability to build multiple platforms
-      # concurrently. It makes the build log unreadable but that's OK
-      # because we have other, slower actions with readable logs.
+      # concurrently. It makes the build log unreadable, so retry with
+      # a single thread in case of failure
       - name: build all and stage
-        run: ./scripts/docker-run.sh make -j3 -C installer/ tarball
+        run: ./scripts/docker-run.sh make -j3 -C installer/ tarball ||
+            VERBOSE=1 NO_PROCESSORS=1 USE_XARGS=no
```
Comment (cujomalainey) [inline .github/workflows/installer.yml:34]: NO_PROCESSOR might be confusing, maybe NUM_THREADS?

Comment (marc-hb) [reply .github/workflows/installer.yml:34]: I don't like it either but it's been part of the .sh scripts "API" for a long time now.

Comment (cujomalainey) [reply .github/workflows/installer.yml:34]: Ack

**Pull request merged**